pallets/xyk/src/lib.rs

```diff
@@ -139,47 +154,48 @@ decl_module! {
                 (second_asset_id, first_asset_id), second_asset_amount
             );
 
-            let liquidity_asset_id = <assets::Module<T>>::assets_next_asset_id();
+
+            let liquidity_asset_id = T::Currency::create(&sender, initial_liquidity);
 
             <LiquidityAssets<T>>::insert((first_asset_id, second_asset_id), liquidity_asset_id);
             <LiquidityPools<T>>::insert(liquidity_asset_id, (first_asset_id, second_asset_id));
 
-            let initial_liquidity = first_asset_amount + second_asset_amount;
-            <assets::Module<T>>::assets_issue(
-                &sender,
-                &initial_liquidity
-            );
-
-            <assets::Module<T>>::assets_transfer(
-                &first_asset_id,
+            T::Currency::transfer(
+                first_asset_id,
                 &sender,
                 &vault,
-                &first_asset_amount
+                first_asset_amount,
+                ExistenceRequirement::KeepAlive
```

Comment:
maybe change all ExistenceRequirement to AllowDeath, as transfer in ORML does allow death

Reply:
not that it matters as the argument in ignored in ORML.

Reply:
is that expected behaviour to remove account when balance is 0 ?

Reply:
agreed to change to `AllowDeath`